### PR TITLE
Add jshint Testing

### PR DESCRIPTION
**Pull request by Gomez**
This adds npm & jshint to travis and fixes the js problems.

@DeepDiver1975 @jancborchardt

### CODE REVIEW

**Comment by jancborchardt** [comment]
Check it @wurstchristoph @clementhk @Xenopathic 

(@Gomez if you put the checkboxes in a bulleted list they become clickable ;)

**Comment by jancborchardt** [comment]
@Gomez also, 0.1 or 0.2?

**Comment by Gomez** [comment]
I would say 0.1 as it should not break anything. Will do baby steps, first jshint, some boilerplate, some tests..
Bower refactoring of existing stuff will be a extra pr.
(bulleted list, right. Was the plan :))

**Comment by DeepDiver1975** [comment]
Nice!

File: package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "mail",
+  "version": "0.0.1",
```
Comment (Xenopathic) [inline package.json]: Version should be 0.1?

Comment (Gomez) [inline package.json]: I used "npm init" to create the skeleton. 0.1 showed a error. Don't know if it is important.

**Comment by Gomez** [comment]
Depend on https://github.com/owncloud/mail/issues/558 before merge

**Comment by DeepDiver1975** [comment]
@Gomez I'm all in to merge this right away - open topics can be fixed in follow up PRs - objections?

**Comment by Gomez** [comment]
Will split the tickets and make some more refactoring notes in the code.
Than ready to merge

**Comment by DeepDiver1975** [comment]
needs rebase as well ....

**Comment by clementhk** [comment]
LGTM

**Comment by Gomez** [comment]
Travis passing, ready to merge?

File: js/mail.js [hunk 2]
```diff
@@ -85,7 +89,8 @@ var Mail = {
 									// If it's okay let's create a notification
 									var tag = 'not-' + f.accountId + '-' + f.name;
 									var icon = OC.filePath('mail', 'img', 'mail-notification.png');
-									Mail.BackGround.showNotification(localAccount.get('email'), body, tag, icon, f.accountId, f.id);
+									Mail.BackGround.showNotification(localAccount.get('email'), body, tag, icon,
+                                                                     f.accountId, f.id);
```
Comment (Xenopathic) [inline js/mail.js]: I've spotted some spaces used for indentation! :speak_no_evil:

File: js/views/folder.js [hunk 3]
```diff
@@ -93,9 +93,11 @@ views.Folders = Backbone.Marionette.CollectionView.extend({
 		var unread = activeFolder.get('unseen');
 
 		if ( unread > 0) {
-			window.document.title = activeFolder.get('name') + ' (' + unread + ') - ' + activeAccount.get('email') + ' - Mail - ' + oc_defaults.title;
+			window.document.title = activeFolder.get('name') + ' (' + unread + ') - ' +
+                                    activeAccount.get('email') + ' - Mail - ' + oc_defaults.title;
```
Comment (Xenopathic) [inline js/views/folder.js]: Spaces! Nooo!

File: js/views/folder.js [hunk 4]
```diff
@@ -93,9 +93,11 @@ views.Folders = Backbone.Marionette.CollectionView.extend({
...
 		} else {
-			window.document.title = activeFolder.get('name') + ' - ' + activeAccount.get('email') + ' - Mail - ' + oc_defaults.title;
+			window.document.title = activeFolder.get('name') + ' - ' + activeAccount.get('email') +
+                                    ' - Mail - ' + oc_defaults.title;
```
Comment (Xenopathic) [inline js/views/folder.js]: More spaces!

File: package.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  "author": "Thomas Müller and others",
+  "license": "AGPL",
+  "bugs": {
+    "url": "https://github.com/owncloud/mail/issues"
+  },
+  "homepage": "https://github.com/owncloud/mail",
+  "devDependencies": {
+    "jshint": "^2.7.0",
+    "jsxhint": "^0.14.0"
+  }
```
Comment (Xenopathic) [inline package.json]: This entire file is space indented!

Comment (Gomez) [inline package.json]: :( Back to "Getting started with PHPStrom" :)

**Comment by Xenopathic** [comment]
@Gomez how goes the indentation fixes? And, could you squash your commits into perhaps two (one for fixes, one for the JSHint stuff)

**Comment by DeepDiver1975** [comment]
@Gomez squashed and spaces fixed for you

**Comment by DeepDiver1975** [comment]
````
$ npm test

> mail@0.0.1 test /home/travis/build/owncloud/core/apps/mail

> jshint . && ./node_modules/.bin/jsxhint .

js/send-mail.js: line 76, col 13, Bad line breaking before '+'.

js/send-mail.js: line 77, col 13, Bad line breaking before '+'.

js/send-mail.js: line 78, col 13, Bad line breaking before '+'.

js/settings.js: line 1, col 1, 'Handlebars' is defined but never used.

js/views/message.js: line 121, col 25, Bad line breaking before '+'.

js/views/message.js: line 122, col 25, Bad line breaking before '+'.

js/views/message.js: line 123, col 25, Bad line breaking before '+'.

7 errors

npm ERR! Test failed. See above for more details.

**Comment by Gomez** [comment]
@DeepDiver1975 Thanks for fixing. I will look at the remaining errors...

**Pull request merged**